### DISCUSSION TITLE
How to use DSPs in FINN?

### DISCUSSION THREAD

**Discussion by MFJI** [Xilinx/finn#888] [Q&A] [answered]
Hello everyone,

I would like to know how I can decide whether DSPs are used or not. Thank you.

**Comment by fpjentzsch** [comment, marked as answer]
Hi,

for the MatrixVectorActivation layer, you can set the `resType` attribute to "dsp": https://github.com/Xilinx/finn/blob/13313c293eb00d2eb88edcaf68386fdf52152aeb/src/finn/custom_op/fpgadataflow/matrixvectoractivation.py#L70C14-L70C21

This should instruct Vitis HLS to use DSPs for the MAC operations. A more efficient implementation is work-in-progress (https://github.com/Xilinx/finn/pull/794).
